File: Android/Tejash11_TejashSeth_2024csit1065_2/FunctionAndArrays/ReverseAnArray.java
import java.io.*;
import java.util.*;

public class Main{
  public static void display(int[] a){
    StringBuilder sb = new StringBuilder();

    for(int val: a){
      sb.append(val + " ");
    }
    System.out.println(sb);
  }

  public static void reverse(int[] a)
  {
    for(int i=0;i<a.length/2;i++)
    {
    int temp = a[i];
    a[i]=a[a.length-i-1]; 
    a[a.length-i-1]=temp;
    }
  } 

public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    int n = Integer.parseInt(br.readLine());
    int[] a = new int[n];
    for(int i = 0; i < n; i++){
       a[i] = Integer.parseInt(br.readLine());
    }

    reverse(a);
    display(a);
 }

}

File: ML/visheshks04_VisheshKumarSingh_2024csit1105_2/FunctionAndArrays/ReverseAnArray.cpp
#include<iostream>
using namespace std;

void reverse(int* arr, int n){
    
    int temp;
    for(int i=0;i<n/2;i++){
        temp = arr[i];
        arr[i] = arr[n-i-1];
        arr[n-i-1] = temp;
    }
}

void display(int* arr, int n){
    for(int i = 0 ; i < n; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

int main(){
    int n;
    cin>>n;
    
    int* arr = new int[n];
    for(int i = 0 ; i < n; i++){
        cin>>arr[i];
    }
    reverse(arr,n);
    display(arr,n);
}
